.github/workflows/build.yml

```diff
@@ -16,28 +16,31 @@ jobs:
         with:
           submodules: recursive
 
-      - name: Setup node
-        uses: actions/setup-node@v3
-        with:
-          node-version: 18.x
-          cache: "yarn"
-
       - name: Install Foundry
         uses: foundry-rs/foundry-toolchain@v1
         with:
           version: nightly
 
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8.x
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+          cache: pnpm
```

Comment:
the order of these steps matters, it needs to be:
- checkout (so pnpm lock file is present)
- pnpm setup (so next step can use pnpm to determine cache location)
- node setup

will clean this up once https://github.com/latticexyz/mud/pull/588 lands